### ISSUE TITLE
Re-add rgba compatibility for text (6.0 had it)

### ISSUE THREAD

**Issue by treboralmasy** [mpdf/mpdf#1387]
# Feature request

Up to mpdf 6.0 opacity was an allowed attribute in text. Ie:
`<span style="color:rgba(255,0,0,0.5)">I'm halfway opac</span>`

This was very handy for halving certain colors over watermarks and would be great to have back.

**Comment by finwe** [comment]
It should be possible even now, so this is probably a regression. Marking as a bug.

**Comment by treboralmasy** [comment]
Maybe, I know the manual is saying that it only allowed in BGs and not in text.

**Comment by finwe** [comment]
Will investigate.

**Comment by finwe** [comment]
6.0 also does not support opacity for text.

**Comment by treboralmasy** [comment]
Yes it does.
Specifically the code for it is located on line 4528.

```
		// mPDF 6.0.1
		$textColorAlpha .= '';
		$Qset = false;
		if($this->ColorFlag) { 
			$textColorAlpha .=' '.$this->TextColor.' ';
			$Qset = true;
		}
		if ($this->colorarray[0]==5) {	// RGBa
			$textColorAlpha .= ' '.$this->SetAlpha(ord($this->colorarray[4])/100, 'Normal', true, 'B').' '; 
			$Qset = true;
		}
		else if ($this->colorarray[0]==6) {	// CMYKa
			$textColorAlpha .= ' '.$this->SetAlpha(ord($this->colorarray[5])/100, 'Normal', true, 'B').' ';
			$Qset = true;
		}

		if ($Qset) { $s .= 'q' . $textColorAlpha; }
```

And here is some code to create on with an alpha. 

`	
	error_reporting(0);
	ini_set('display_errors', 0);	
	include_once($_SERVER['DOCUMENT_ROOT']."/mpdf60/mpdf.php");
	$mpdf=new mPDF();
	$mpdf->WriteHTML( '<div style="color:rgba(255,255,255,0)">hello</div>');
	$mpdf->Output();
`

Though I may be mistaken on the functionality. It appears as though it support 0 and 1 (nothing in between). 
Still it is useful to have 0 opacity as it will allow watermarks to show below, while a white font obviously will not.

**Comment by finwe** [comment]
No, I have just tested a text in 6.0 with your code and the opacity was not applied. If the documentation says opacity is not supported on text, then this is a feature request.

**Comment by treboralmasy** [comment]
Here is a link to the pdf that generates.
https://filebin.net/xzjud8fu76re3voc/mpdf.pdf?t=plg7mi1l
As you can see it is transparent.

**Comment by finwe** [comment]
255,255,255 is white. Black is still displayed as black no matter the opacity.

**Comment by treboralmasy** [comment]
Okay. Bad example. How about this then.
`
	error_reporting(0);
	ini_set('display_errors', 0);	
	include_once($_SERVER['DOCUMENT_ROOT']."/code_worksheets/mpdf60/mpdf.php");
	$mpdf=new mPDF();
	$mpdf->WriteHTML( 'I am normal text. <span style="color:rgba(255,0,0,0)">I am transparent.</span> I am normal again.');
	$mpdf->Output();
`
And here is that pdf.
https://filebin.net/a2ws9cpjd1t5c1tn/mpdf__1_.pdf?t=ey2021zf

**Comment by finwe** [comment]
Unable to reproduce, I have tried mPDF 6.0 from both original ZIP and composer, 6.1 from composer and 8.0 and all files render the same.
 
[pdf.pdf](https://github.com/mpdf/mpdf/files/6016431/pdf.pdf)

**Comment by treboralmasy** [comment]
Don't know. I got mine from Ian's site way back in the day. Works on mine (actually it does allow values between 0 and 1, I just forgot I was using white).
Here is a comparison between the mpdf 6.0 I was able to find and my version (the latest 6.0)
https://www.diffchecker.com/1JqFqzpb

**Comment by finwe** [comment]
Can you share the whole 6.0.1 archive, please?

**Comment by treboralmasy** [comment]
And here is the mpdf file itself:
`https://filebin.net/1yk4fz988t28g29u/mpdf.php?t=2fdeotj4`

**Comment by treboralmasy** [comment]
This is the only part that effects transparency. The entire archive is too large (34mb).

**Comment by treboralmasy** [comment]
Also though, you do need a file in the classes called mongolian.php or it won't work with the current mpdf 6.0

**Comment by finwe** [comment]
Can you share the whole 6.0.1 archive somehow even though it is too large, please?

It seems there are changes that are not applied in the project at github.

**Comment by treboralmasy** [comment]
What is your email? I can email it to you.

**Comment by finwe** [comment]
34mb is too large for an e-mail. Use a filesharing service such as https://wetransfer.com/

**Comment by finwe** [comment]
Anyway, this fork of the project never supported text opacity, so this remains a feature request.

**Comment by treboralmasy** [comment]
https://we.tl/t-jPWW5hhoGS

**Comment by finwe** [comment]
Thanks.

**Comment by mprythero** [comment]
Has there been any update on this?

**Comment by finwe** [comment]
No.
